les/costtracker.go

```diff
@@ -224,13 +263,16 @@ func (ct *costTracker) gfLoop() {
 						ct.gfLock.Unlock()
 						if ch != nil {
 							select {
-							case ct.totalRechargeCh <- uint64(ct.utilTarget * gf):
+							case ct.totalRechargeCh <- uint64(totalRecharge):
 							default:
 							}
 						}
 						log.Debug("global cost factor updated", "gf", gf, "weight", time.Duration(gfWeight))
 					}
 				}
+				ct.logRecentUsage.Update(gfUsage)
+				ct.logTotalRecharge.Update(totalRecharge)
+
 			case stopCh := <-ct.stopCh:
```

Comment:
Can we also add a ticker here to persist `gfSum` and `gfWeight` at a constant interval?

Reply:
done